ipn/ipnlocal/local.go

```diff
@@ -1935,14 +1935,27 @@ func normalizeResolver(cfg dnstype.Resolver) dnstype.Resolver {
 	return cfg
 }
 
-// tailscaleVarRoot returns the root directory of Tailscale's writable
+// TailscaleVarRoot returns the root directory of Tailscale's writable
 // storage area. (e.g. "/var/lib/tailscale")
-func tailscaleVarRoot() string {
+//
+// It returns an empty string if there's no configured or discovered
+// location.
+func (b *LocalBackend) TailscaleVarRoot() string {
 	switch runtime.GOOS {
 	case "ios", "android":
 		dir, _ := paths.AppSharedDir.Load().(string)
 		return dir
 	}
+	// Temporary (2021-09-27) transitional fix for
+	// #2927 (Synology cert dir) on the way towards
+	// a more complete (#2932).
+	if fs, ok := b.store.(*ipn.FileStore); ok {
```

Comment:
Is this safe to do generally? Might trigger on non-synology devices?

Reply:
that's intended. It's not a Synology-only fix.